test/specs/EventsSpec.js

```diff
@@ -21,325 +21,355 @@ describe("Event Tests", function() {
           testSlider.data('slider')._mouseup(mouse);
-          expect(flag).not.toBeTruthy();
+          window.setTimeout(function() {
+            expect(spy).not.toHaveBeenCalled();
+            done();
+          });
         });
       });
 
     });
 
     describe("Touch Events", function() {
       var touch;
+      var spy;
 
       beforeEach(function() {
         touch = document.createEvent('Event');
         var dummyTouchEvent = document.createEvent('MouseEvents');
         touch.touches = [dummyTouchEvent];
         window.ontouchstart = true;
+        spy = jasmine.createSpy('spy');
       });
 
       afterEach(function() {
         window.ontouchstart = null;
       });
 
-      it("'slideStart' event is triggered properly and can be binded to", function() {
+      it("'slideStart' event is triggered properly and can be binded to", function(done) {
         touch.initEvent("touchstart");
-
-        testSlider.on('slideStart', function() {
-          flag = true;
-        });
+        testSlider.on('slideStart', spy);
         testSlider.data('slider')._mousedown(touch);
-
-        expect(flag).toBeTruthy();
+        window.setTimeout(function() {
+          expect(spy).toHaveBeenCalled();
+          done();
+        });
       });
 
-      it("'slide' event is triggered properly and can be binded to", function() {
+      it("'slide' event is triggered properly and can be binded to", function(done) {
         touch.initEvent("touchmove");
-
-        testSlider.on('slide', function() {
-          flag = true;
-        });
+        testSlider.on('slide', spy);
         testSlider.data('slider')._mousemove(touch);
-
-        expect(flag).toBeTruthy();
+        window.setTimeout(function() {
+          expect(spy).toHaveBeenCalled();
+          done();
+        });
       });
 
-      it("'slide' event sets the right value on the input", function() {
+      it("'slide' event sets the right value on the input", function(done) {
         touch.initEvent("touchmove");
 
         testSlider.on('slide', function() {
-          flag = true;
           expect(isNaN(testSlider.val())).not.toBeTruthy();
+          done();
         });
         testSlider.data('slider')._mousemove(touch);
-
-        expect(flag).toBeTruthy();
       });
 
-      it("'slide' event value and input value properties are synchronous", function() {
+      it("'slide' event value and input value properties are synchronous", function(done) {
         touch.initEvent("touchmove");
 
         testSlider.on('slide', function(e) {
-          flag = true;
           expect(e.value.toString()).toEqual(testSlider.val());
+          done();
         });
         testSlider.slider("setValue", 3, true, false);
-
-        expect(flag).toBeTruthy();
       });
 
-      it("'change' event value and input value properties are synchronous", function() {
+      it("'change' event value and input value properties are synchronous", function(done) {
```

Comment:
This test is testing whether the `change` event is triggered synchronously. The `done` callback should be removed.